### PR TITLE
Add wait times.

### PR DESCRIPTION
**Pull request by YKo20010**
## Overview
Added MPAndroidChart Bar Graph and stylized as per design.

## Next Steps (delete if not applicable)
Need to add actual wait-time data to graph.

## Screenshots
image

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java [hunk 1]
```diff
@@ -207,6 +226,109 @@ else if (mCafeData instanceof DiningHallModel) {
                         ContextCompat.getColor(getApplicationContext(), R.color.blue));
             }
         }
+
+        setupWaitTimesChart();
+    }
+
+    private void setupAxis(BarChart b) {
+        // Remove bar chart Y axis - left side.
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: nice comments!

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java [hunk 2]
```diff
@@ -207,6 +226,109 @@ else if (mCafeData instanceof DiningHallModel) {
...
+        xAxis.setLabelCount(8);
+        xAxis.setTypeface(Typeface.SANS_SERIF);
+        xAxis.setYOffset(8f);
+        ArrayList<String> xAxisLabels = new ArrayList<String>();
+        for(int i = 0; i < 22; i++) {
+            int hourNum = (i + 6) % 12;
+            if (hourNum == 0) {
+                hourNum = 12;
+            }
+            String ap = i <= 5 || i >= 18 ? "a" : "p";
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: i'm not a huge fan of the am/pm label on the graphs. is that what the design shows?

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java [hunk 3]
```diff
@@ -207,6 +226,109 @@ else if (mCafeData instanceof DiningHallModel) {
...
+        mWaitTimesChart = (BarChart) findViewById(R.id.wait_time_chart);
+        mWaitTimesChart.setDescription(null);
+        mWaitTimesChart.getLegend().setEnabled(false);
+        mWaitTimesChart.setDoubleTapToZoomEnabled(false);
+        mWaitTimesChart.setPinchZoom(false);
+        mWaitTimesChart.animateY(2000);
+        mWaitTimesChart.setExtraBottomOffset(24f);
+        mWaitTimesChart.getRendererXAxis().getPaintAxisLabels().setTextAlign(Paint.Align.LEFT);
+
+
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: remove newline

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java [hunk 4]
```diff
@@ -207,6 +226,109 @@ else if (mCafeData instanceof DiningHallModel) {
...
+        mWaitTimesChart.setOnChartValueSelectedListener(new OnChartValueSelectedListener() {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: these overridden methods are empty?

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java [hunk 5]
```diff
@@ -207,6 +226,109 @@ else if (mCafeData instanceof DiningHallModel) {
...
+        entries.add(new BarEntry(11f, 5f));
+        entries.add(new BarEntry(12f, 10f));
+        entries.add(new BarEntry(13f, 60f));
+        entries.add(new BarEntry(14f, 40f));
+        entries.add(new BarEntry(15f, 20f));
+        entries.add(new BarEntry(16f, 2f));
+        entries.add(new BarEntry(17f, 2f));
+        entries.add(new BarEntry(18f, 10f));
+        entries.add(new BarEntry(19f, 2f));
+        entries.add(new BarEntry(20f, 2f));
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: please use a loop

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java [hunk 6]
```diff
@@ -207,6 +226,109 @@ else if (mCafeData instanceof DiningHallModel) {
...
+
+        BarDataSet set = new BarDataSet(entries, "BarDataSet");
+        set.setColors(ContextCompat.getColor(getApplicationContext(), R.color.lightblue));
+        set.setValueTypeface(Typeface.SANS_SERIF);
+//        set.setHighlightEnabled(false);
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: remove or add a `TODO`

File: app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java [hunk 7]
```diff
@@ -137,6 +140,7 @@ public void onClick(View v) {
 
     public void initializeCampusEateryButtons(View rootView) {
         int[] viewIds = {
+//                R.id.nearestFirstButton,
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java]: remove or add a `TODO`

File: app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java [hunk 8]
```diff
@@ -374,6 +387,11 @@ private void updateListAdapter() {
 
     @Override
     public void onClick(View view) {
+//        if (view.getId() == R.id.nearestFirstButton) {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java]: remove or add a `TODO`

File: app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java [hunk 9]
```diff
@@ -374,6 +387,11 @@ private void updateListAdapter() {
...
+//            mFirebaseAnalytics.logEvent("nearest_first_filter_press", null);
+//            mNearestFirstButtonPressed = !mNearestFirstButtonPressed;
+//            handleNearestFirstButtonPress(mCampusButtons.get(view.getId()));
+//        }
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/eatery/MainListFragment.java]: remove or add a `TODO`

File: app/src/main/res/layout/activity_campus_eatery.xml [hunk 10]
```diff
@@ -142,6 +142,59 @@
...
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/infobox"
+                android:background="@color/white"
+                android:elevation="1dp"
+                android:paddingTop="7dp"
+                android:paddingBottom="7dp"
+                android:paddingRight="16dp"
+                android:paddingLeft="16dp"
+                >
```
Comment (ningning621) [inline app/src/main/res/layout/activity_campus_eatery.xml]: move to previous line

File: app/src/main/res/layout/activity_campus_eatery.xml [hunk 11]
```diff
@@ -142,6 +142,59 @@
...
+                <ImageView
+                    android:layout_width="match_parent"
+                    android:layout_height="1dp"
+                    android:background="@color/inactive"
+                    />
```
Comment (ningning621) [inline app/src/main/res/layout/activity_campus_eatery.xml]: move to previous line

File: app/src/main/res/layout/activity_campus_eatery.xml [hunk 12]
```diff
@@ -142,6 +142,59 @@
...
+                android:id="@+id/wait_time_graph"
+                android:layout_width="fill_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/wait_time_label"
+                android:background="@color/white">
+                <com.github.mikephil.charting.charts.BarChart
+                    android:id="@+id/wait_time_chart"
+                    android:layout_width="match_parent"
+                    android:layout_height="180dp"
+                    />
```
Comment (ningning621) [inline app/src/main/res/layout/activity_campus_eatery.xml]: move to previous line

File: app/src/main/res/layout/fragment_main_list.xml [hunk 13]
```diff
@@ -24,6 +24,25 @@
                 android:background="@color/white"
                 android:orientation="horizontal"
                 android:paddingBottom="8dp">
+<!--                <Button-->
```
Comment (ningning621) [inline app/src/main/res/layout/fragment_main_list.xml]: remove or add a `TODO`

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java (hunk 5 above)
Comment (ningning621) [reply app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: oh wait this is temporary data, i get it now. 

please add a comment saying that this is temporary, and add something like `TODO (yanlam): pull wait times from backend and make bar entries dynamic`

File: app/src/main/res/values/colors.xml [hunk 14]
```diff
@@ -3,6 +3,7 @@
 	<color name="colorPrimaryDark">#476693</color>
 	<color name="blue">#4a90e2</color>
     <color name="fadedblue"> #994a90e2</color>
+	<color name="lightblue">#b7d3f3</color>
```
Comment (young-k) [inline app/src/main/res/values/colors.xml]: can you fix the indentation in the last color, and fix the casing so it's consistent? `colorPrimaryDark` is camelcased, but nothing else seems to be?

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java (hunk 2 above)
Comment (YKo20010) [reply app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: Yeah, this is the design:

image

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java (hunk 4 above)
Comment (YKo20010) [reply app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: They will be updated later.

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java (hunk 2 above)
Comment (ningning621) [reply app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: aight.

File: app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java (hunk 4 above)
Comment (ningning621) [reply app/src/main/java/com/cornellappdev/android/eatery/CampusMenuActivity.java]: please leave a `TODO` here

**Pull request merged**